Classes/Hooks/HookService.php

```diff
@@ -0,0 +1,114 @@
+<?php
+declare(strict_types=1);
+
+namespace OliverKlee\Seminars\Hooks;
+
+use UnexpectedValueException;
+
+use OliverKlee\Seminars\Interfaces\Hook;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+/**
+ * This class provides functions for unified hooks.
+ *
+ * It unifies the wide-spread use of this type of hooks in seminars.
+ *
+ * @author Michael Kramer <m.kramer@mxp.de>
+ */
+class HookService
+{
+    /**
+     * Interface name to this hook
+     *
+     * @var string
+     */
+    protected $interfaceName = '';
+
+    /**
+     * Index in $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'] of hooked-in classes
+     *
+     * @var string
+     */
+    protected $index = '';
+
+    /**
+     * Hook objects built
+     *
+     * @var array
+     */
+    protected $hookObjects = [];
+
+    /**
+     * @var boolean
+     */
+    protected $hooksHaveBeenRetrieved = false;
+
+    /**
+     * The constructor.
+     *
+     * @param $interfaceName interface the hook needs implemented
+     * @param $index optional index to $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars']
+     *               if not using the interface name (for backwards compatibility)
+     *               (the interface name is recommended)
+     *
+     * @throws \UnexpectedValueException
+     *         if $interfaceName does not extend Hook interface
```

Comment:
Please provide the verbatim interface name in backticks here (without the namespace) so the reader knows which exact interface the comment refers to (or drop the comment, which would be fine for me as well).